src/dbus/service.rs

```diff
@@ -18,24 +18,35 @@ use crate::Result;
 /// ```
 pub struct Service<'a> {
     inner: Arc<api::Service<'a>>,
-    #[allow(unused)]
-    service_key: Option<Vec<u8>>,
+    aes_key: Option<Arc<Key>>,
     session: Arc<api::Session<'a>>,
-    algorithm: Arc<Algorithm>,
+    algorithm: Algorithm,
 }
 
 impl<'a> Service<'a> {
     /// Create a new instance of the Service.
     pub async fn new(algorithm: Algorithm) -> Result<Service<'a>> {
         let cnx = zbus::Connection::session().await?;
         let service = Arc::new(api::Service::new(&cnx).await?);
-        let (service_key, session) = service.open_session(&algorithm).await?;
+
+        let private_key = Key::generate_private_key();
```

Comment:
You don't need this if it Algorithm::Plain, so I would move it to inside the match and make it return a tuple (public_key, private_key)